src/HandleMgr.cpp

```diff
@@ -78,6 +79,16 @@ namespace splash
         this->fileAccProperties = fileAccProperties;
         this->fileFlags = flags;
         this->singleFile = false;
+        // Validation: For parallel files we normally append MPI rank or iteration number.
+        //             This is disabled by using FNS_FULLNAME
+        //             or when the filename already contains an h5-extension.
+        if (fileNameScheme != FNS_FULLNAME && baseFilename.find(".h5") != std::string::npos)
+        {
+            log_msg(1, "\n"
```

Comment:
not 100% satisfied. If any component in `mpiSize` is > 1 this should simply fail and be disallowed.